GitUI/CommandsDialogs/SettingsDialog/Pages/DetailedSettingsPage.cs

```diff
@@ -36,13 +36,22 @@ protected override void Init(ISettingsPageHost pageHost)
 
         protected override void PageToSettings()
         {
-            AppSettings.SmtpServer = SmtpServer.Text;
-            if (int.TryParse(SmtpServerPort.Text, out var port))
+            if (CurrentSettings is ISettingsSource settingsSource)
             {
-                AppSettings.SmtpPort = port;
-            }
+                var detailedSettings = settingsSource.Detailed();
+
+                detailedSettings.SmtpServer = SmtpServer.Text;
+
+                if (int.TryParse(SmtpServerPort.Text, out var port))
+                {
+                    detailedSettings.SmtpPort = port;
+                }
 
-            AppSettings.SmtpUseSsl = chkUseSSL.Checked;
+                detailedSettings.SmtpUseSsl = chkUseSSL.Checked;
+
+                SmtpServerPort.Text = detailedSettings.SmtpPort.ToString();
+                chkUseSSL.Checked = detailedSettings.SmtpUseSsl;
```

Comment:
Any reason for this?
(at least checked should be redundant)

Reply:
Existing code.

Reply:
What would you like to change?

Reply:
👍